doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+of the OpenCV library source tree.
+
+It is a well-known fact that geterogeneous algorithms based on Computer Vision
+image processing with the backup of deep neural networks inference
+(Deep Learning) are able to solve a huge variety of problems which CV-only
+ones might experience difficulties with. Let's consider the certain
+algorithm as an example.
+
+# Face Beautification algorithm {#gapi_fb_algorithm}
+
+The algorithm of retouching a face has been chosen as the subject for this
+tutorial. Its main idea is to implement a real-time
+videostream processing that detects faces and applies some filters to make them
+look beautiful (more or less). The pipeline is the following:
+
+![The Face Beautification algorithm](pics/algo.png)
+
+The algorithm consists of two parts: an inference of two networks (including
+data pre- and post-processing) and an image filtering pipeline which uses the
+inference data to create masks, then applies filters and composes the output
+retouched image. What is each of these parts for? To be more certain, the
+"beautification" implemented here is smoothing out the skin (by the
+`Bilateral filter`) and making eyes, a nose and a mouth sharper (by the
+`Unsharp mask`). Thus, to know which filter should be applied to which area of
+image, we need to calculate a contour of the face and contours of facial
+elements --- we can get them from data received via deep learning methods.
+
+Two topologies from OMZ have been used in this sample: the
+<a href="https://github.com/opencv/open_model_zoo/tree/master/models/intel
+/face-detection-adas-0001">face-detection-adas-0001</a>
+and the
+<a href="https://github.com/opencv/open_model_zoo/blob/master/models/intel
+/facial-landmarks-35-adas-0002/description/facial-landmarks-35-adas-0002.md">
+facial-landmarks-35-adas-0002</a>.
+
+The face detector takes the input image and returns a blob with the shape
+[1, 1, N, 7] after the inference, where `N = 200` is the maximum number of
+faces which can be detected. Obviously, the output `cv::Mat` post-processing is
+required. The structure of the output for every face is the following:
+`[image_id, label, conf, x_min, y_min, x_max, y_max]`,
+all the seven elements are floating point. For our further goals we need to take
+only results with `image_id > 0` because a negative value in this field
+indicates the end of detections; also we can cut detections by the `conf` field
+to avoid mistakes of the detector. The last four float numbers we've got denote
+normalized coordinates and are between 0 and 1; to get the real pixel
+coordinates we should multiply it by the image sizes respectively to the
+directions.
+
+Then, to create masks we need to get landmarks of facial elements for every
+detected face. The chosen detector expects to obtain square area with only one
+face contained so we have to infer the model separately for every ROI we've
+got from the previous detection. The output of the facial landmarks detector is
```

Comment:
"Face masks are built using facial landmarks, which are inferred for every face".